apps/api/pkg/handlers/tenant.go

```diff
@@ -85,38 +98,72 @@ func getTenantHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func updateTenantHandler(w http.ResponseWriter, r *http.Request) {
-	var payload struct {
-		Name string `json:"name"`
-		Id   string `json:"id"`
+	type Payload struct {
+		TenantId          int64  `json:"tenant_id"`
+		TenantName        string `json:"tenant_name"`
+		RequestedByUserId string `json:"requested_by_id"`
 	}
+
+	var payload Payload
+
 	if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
 		http.Error(w, "Invalid request payload", http.StatusBadRequest)
 		return
 	}
 
-	err := database.UpdateTenantName(database.DB, payload.Id, payload.Name)
+	requestedByUser, err := database.GetTenantUser(payload.RequestedByUserId)
+	if err != nil {
+		http.Error(w, "Failed to get user", http.StatusInternalServerError)
+		return
+	}
+
+	if requestedByUser.Role != "admin" && requestedByUser.Role != "writer" && requestedByUser.TenantId != payload.TenantId {
+		http.Error(w, "User is not authorized to perform this action", http.StatusUnauthorized)
+		return
+	}
+
+	err = database.UpdateTenantName(database.DB, payload.TenantId, payload.TenantName)
```

Comment:
UpdateTenantName is only do one function. We have to manage UpdateTenant because we will have multiple values in future. We have to manage all in one.